cdap-ui/app/cdap/components/AbstractWidget/InputKeyValueDropdown/InputKeyValueDropdown.tsx

```diff
@@ -0,0 +1,181 @@
+import { IStageSchema, IWidgetProps } from 'components/AbstractWidget';
+import { objectQuery } from 'services/helpers';
+import Input from '@material-ui/core/Input';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import withStyles from '@material-ui/core/styles/withStyles';
+
+const styles = (theme) => {
+  return {
+    ...AbstractRowStyles(theme),
+    inputContainer: {
+      display: 'grid',
+      gridTemplateColumns: '50% 50%',
+      gridGap: '10px',
+    },
+    disabled: {
+      color: `${theme.palette.grey['50']}`,
+    },
+  };
+};
+
+interface IField {
+  name: string;
+  type: string[];
+}
+
+interface IComplexDropdown {
+  value: string | number;
+  label: string;
+}
+
+export type IDropdownOption = string | number | IComplexDropdown;
+
+interface IInputKeyValueDropdownProps extends IAbstractRowProps<typeof styles> {
+  keyPlaceholder?: string;
+  kvDelimiter?: string;
+  // dropdownOptions: IDropdownOption[];
+  dropdownOptions: any;
+  extraConfig?: any;
+}
+
+interface IKeyValueState {
+  value: string;
+  key: string;
+}
+
+type StateKeys = keyof IKeyValueState;
+
+class InputKeyValueDropdown extends AbstractRow<IInputKeyValueDropdownProps, IKeyValueState> {
+  public static defaultProps = {
+    keyPlaceholder: 'Test',
+    kvDelimiter: '=',
+    dropdownOptions: [],
+    delimiter: ',',
+  };
+
+  public state = {
+    key: '',
+    value: '',
+  };
+
+  public componentDidMount() {
+    const [key = '', value = ''] = this.props.value.split(this.props.kvDelimiter);
+
+    this.setState({
+      key,
+      value,
+    });
+  }
+
+  private handleChange = (type: StateKeys, e) => {
+    this.setState(
+      {
+        [type]: e.target.value,
+      } as Pick<IKeyValueState, StateKeys>,
+      () => {
+        const key = this.state.key;
+        const value = this.state.value;
+
+        const updatedValue = key.length > 0 ? [key, value].join(this.props.kvDelimiter) : '';
+        this.onChange(updatedValue);
+      }
+    );
+  };
+
+  public getFields(schemas: IStageSchema[], allowedTypes: string[]) {
+    let fields = [];
+    if (!schemas || schemas.length === 0) {
+      return fields;
+    }
+    const stage = schemas[0];
+
+    try {
+      const unparsedFields = JSON.parse(stage.schema).fields;
+
+      if (unparsedFields.length > 0) {
+        fields = unparsedFields
+          .filter((field: IField) => this.containsType(field.type, allowedTypes))
+          .map((field: IField) => field.name);
+      }
+    } catch {
+      // tslint:disable-next-line: no-console
+      console.log('Error: Invalid JSON schema');
+    }
+    return fields;
+  }
+
+  // Function that checks if types contains a type that is in allowedTypes
+  // This is meant to handle nullable fields since a nullable string type is
+  // presented as ['string','null'].
+  public containsType(types: string[], allowedTypes: string[]) {
```

Comment:
Based on the usage of `allowedTypes` seems like both this and the `extractType` function is unnecessary?